e2e/tests/test_e2e.py

```diff
@@ -212,6 +220,61 @@ def test_logical_backup_cron_job(self):
         self.assertEqual(0, len(jobs),
                          "Expected 0 logical backup jobs, found {}".format(len(jobs)))
 
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_node_selector(self):
+        k8s = self.k8s
+
+        # 1. Add a label to a node
+        target_node = k8s.api.core_v1.read_node(name='postgres-operator-e2e-tests-worker2')
+        pg_patch_env_special = {
+            "metadata": {
+                "labels": {
+                    "environment": "special"
+                }
+            }
+        }
+        target_node_name = target_node.metadata.name
+        print ("Patching node [%s] with 'environment:special' label" % (target_node_name))
+        k8s.api.core_v1.patch_node(target_node_name, pg_patch_env_special)
+
+        pg_patch_custom_resource = {
+            "spec": {
+                "nodeSelector": {
+                    "environment": "special"
+                }
+            }
+        }
+        k8s.api.custom_objects_api.patch_namespaced_custom_object(
+            group="acid.zalan.do",
+            version="v1",
+            namespace="default",
+            plural="postgresqls",
+            name="acid-minimal-cluster",
+            body=pg_patch_custom_resource)
+
+        k8s.api.core_v1.delete_namespaced_pod("acid-minimal-cluster-0", "default")
+        k8s.api.core_v1.delete_namespaced_pod("acid-minimal-cluster-1", "default")
+        # Unfortunately necessary to allow time for sync to take place
+        time.sleep(90)
```

Comment:
should be sufficient to delete only the replica pod and use the wait function. Do not kill the master like this.